src/App.tsx

```diff
@@ -14,11 +14,109 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isVisible: boolean,
+  goodsDisplayList: string[],
+  initialList: string[],
```

Comment:
since we never modify the initial array, you are being extra careful adding the same array twice